src/middlewared/middlewared/alert/source/ntp_status.py

```diff
@@ -0,0 +1,162 @@
+from datetime import timedelta
+
+from middlewared.alert.base import Alert, AlertLevel, AlertSource
+from middlewared.alert.schedule import IntervalSchedule
+from middlewared.utils import run
+
+import psutil
+import ntplib
+import time
+import re
+import os
+import logging
+
+logger = logging.getLogger(__name__)
+
+# 30 minutes in seconds
+STARTUP_PERIOD = 30*60
+REQUEST_TIMEOUT = 5
+
+SELECT_STATUS_RE = re.compile('^([x\.\-\+\#\*o]?)')
+SELECT_STATUS = {
+    '':  {'code': 0, 'message': 'sel_reject', 'description': "discarded as not valid (TEST10-TEST13)"},
+    'x': {'code': 1, 'message': 'sel_falsetick', 'description': "discarded by intersection algorithm"},
+    '.': {'code': 2, 'message': 'sel_excess', 'description': "discarded by table overflow (not used)"},
+    '-': {'code': 3, 'message': 'sel_outlier', 'description': "discarded by the cluster algorithm"},
+    '+': {'code': 4, 'message': 'sel_candidate', 'description': "included by the combine algorithm"},
+    '#': {'code': 5, 'message': 'sel_backup', 'description': "backup (more than tos maxclock sources)"},
+    '*': {'code': 6, 'message': 'sel_sys.peer', 'description': "system peer"},
+    'o': {'code': 7, 'message': 'sel_pps.peer', 'description': "PPS peer (when the prefer peer is valid)"},
+}
+
+# type - local, unicast, multicast or broadcast
+TYPE = {
+    'l': 'local',
+    'u': 'unicast',
+    'm': 'multicast',
+    'b': 'broadcast',
+    'p': 'pool'
+}
+
+FIELDS = ['remote', 'refid', 'stratum', 'type', 'when', 'poll', 'reach', 'delay', 'offset', 'jitter']
+FIELDS_CNT = len(FIELDS)
+
+
+def since_boot():
+    return time.time() - psutil.boot_time()
```

Comment:
`time.clock_gettime(5)` (where 5 is `CLOCK_UPTIME`, would be added in Python 3.7) is more precise when NTP adjustments happen

Reply:
Nice to know, but it's not there yet and we don't care much about precision here - just need some grace period for letting NTP to connect and synchronize with it's peers.